File: common/src/main/java/com/rabtman/common/utils/ExceptionUtils.java
package com.rabtman.common.utils;

public class ExceptionUtils {

  public static String errToStr(Exception e) {
    StringBuilder stringBuilder = new StringBuilder();
    stringBuilder.append(e.toString() + "\n");
    StackTraceElement[] stack = e.getStackTrace();
    for (StackTraceElement aStack : stack) {
      stringBuilder.append(aStack.toString() + "\n");
    }
    return stringBuilder.toString();
  }

  public static String errToStr(Throwable t) {
    StringBuilder stringBuilder = new StringBuilder();
    stringBuilder.append(t.toString() + "\n");
    StackTraceElement[] stack = t.getStackTrace();
    for (StackTraceElement aStack : stack) {
      stringBuilder.append(aStack.toString() + "\n");
    }
    return stringBuilder.toString();
  }
}

File: common/src/main/java/com/rabtman/common/utils/ExceptionUtils.kt
package com.rabtman.common.utils

object ExceptionUtils {

    fun errToStr(t: Throwable): String {
        val stringBuilder = StringBuilder()
        stringBuilder.append(t.toString())
        stringBuilder.append("\n")
        val stack = t.stackTrace
        for (aStack in stack) {
            stringBuilder.append(aStack.toString())
            stringBuilder.append("\n")
        }
        return stringBuilder.toString()
    }
}